modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/impl/IngestServiceImpl.java

```diff
@@ -1376,45 +1406,80 @@ private void mergeMediaPackageElements(final MediaPackage mp, final MediaPackage
     }
 
     for (MediaPackageElement element : scheduledMp.getElements()) {
-      // Asset manager media package may have a publication element (for live) if retract live has not run yet
-      if (element.getFlavor() != null
-              && !MediaPackageElement.Type.Publication.equals(element.getElementType())
```

Comment:
TODO: check if it's really possible for a mediapackage element to not have a flavor.